pkg/controller/networkpolicy/clusternetworkpolicy.go

```diff
@@ -131,8 +131,15 @@ func (n *NetworkPolicyController) processClusterNetworkPolicy(cnp *secv1alpha1.C
 	appliedToGroupNamesSet := sets.String{}
 	// Create AppliedToGroup for each AppliedTo present in ClusterNetworkPolicy spec.
 	for _, at := range cnp.Spec.AppliedTo {
-		appliedToGroupNamesSet.Insert(n.createAppliedToGroup(
-			"", at.PodSelector, at.NamespaceSelector, at.ExternalEntitySelector))
+		var atg string
+		if at.Group != "" {
+			atg = n.processAppliedToGroupForCG(at.Group)
```

Comment:
Have we tested out what will happen if a CG with ipBlock is given here? From reading `processAppliedToGroupForCG` I believe an empty string will be returned in this case. The `AppliedToGroups` of this internalNP will become a list with an empty string:
```
ss := sets.String{}
ss.Insert("")
for _, s := range ss.List() {
     fmt.Println("There is an element! %s", s)
}
output: There is an element!
```
I'm not sure if this is something that agent can handle smoothly in cache.go, L775:
```
curSet, exists := c.memberSetByGroup[groupName]
if !exists {
	klog.V(2).Infof("AppliedToGroup %v was not found", groupName)
	return nil, false
}
```

Reply:
actually if you see L140, we check "" before inserting in sets